includes/wikia/upload/WikiaUploadStashFile.class.php

```diff
@@ -33,10 +33,19 @@ function getHashPath() {
 	 * @return String: URL to access thumbnail, or URL with partial path
 	 */
 	public function getThumbUrl( $suffix = false ) {
-		$path = $this->repo->getZoneUrl( 'thumb' ) . '/temp/' . $this->getUrlRel();
-		if ( $suffix !== false ) {
-			$path .= '/' . rawurlencode( $suffix );
+		global $wgEnableVignette;
+
+		if ($wgEnableVignette) {
+			$generator = $this->getUrlGenerator();
+			VignetteRequest::applyLegacyThumbDefinition($generator, $suffix);
```

Comment:
For consistency with the test, should this be `$generator = VignetteRequest::applyLegacyThumbDefinition($generator, $suffix);`?